client/src/components/Form/Reservation/index.js

```diff
@@ -158,7 +164,7 @@ class BookingForm extends React.Component {
               style={{
                 width: 200,
               }}
-              disabled={confirmLoading}
+              disabled={confirmLoading || readOnly}
```

Comment:
instead of repeating this condition everywhere in the component, declare a variable and give it `const disabled = confirmLoading || readOnly`